notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "\n",
+      "    :param config: YAML file or ``dict`` (``None`` => read ``stdin``).\n",
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: An initialized ``YAMLConfig`` object.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.get_yaml_config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "606da5b3-4bff-4148-a9a5-908aa7dd5e8c",
+   "metadata": {},
+   "source": [
+    "`get_yaml_config()` can take input from a Python `dict` or a YAML file like the one below.\n",
+    "<!--cell 4-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "c6e049df-38f6-4879-8e0d-68356226d94b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/get-config.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c2f72448-d35e-4a0b-a371-cb47c7b3338b",
+   "metadata": {},
+   "source": [
+    "Paths to config files can be provided either as a string or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Since `get_yaml_config()` is used here, a `YAMLConfig` object is returned.\n",
+    "<!--cell 6-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cc3020a6-4eb4-4830-9263-a9fc8fac7450",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "config1 = config.get_yaml_config(\n",
+    "    config=Path(\"fixtures/config/get-config.yaml\")\n",
+    ")\n",
+    "print(config1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b7bcd736-ff78-4e8b-957f-b348b812c5f6",
+   "metadata": {},
+   "source": [
+    "Providing a Python `dict` will create a `Config` object with format matching the function used.\n",
+    "<!--cell 8-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "f01ac223-4a02-40ba-822f-8e66ad39f313",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "message:\n",
+      "  greeting: Hi\n",
+      "  recipient: Earth\n"
+     ]
+    }
+   ],
+   "source": [
+    "input_config = {\"message\": {\"greeting\":\"Hi\", \"recipient\":\"Earth\"}}\n",
+    "config2 = config.get_yaml_config(\n",
+    "    config=input_config\n",
+    ")\n",
+    "print(config2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dc745e95-d1ce-435c-a488-13b761979e36",
+   "metadata": {},
+   "source": [
+    "## Config Depth Limitations\n",
+    "\n",
+    "Some config formats have limitations on the depth of their nested configs. For example, Shell configs can only contain key-value pairs at the top level.\n",
+    "<!--cell 10-->"
```

Comment:
Does it make sense to include a link in this sentence that describes sh config files? The same goes for INI below.

Reply:
I added a link for INI configs (https://en.wikipedia.org/wiki/INI_file), but am having trouble finding something that seems suitable for Shell configs. Is there any source that you had in mind or any specific information that I could add there?

Reply:
Hmmmm. That's a great question. Maybe it's too basic to have a link. Just stating that bash variables follow the standard bash syntax `key=variable` would be enough?

Reply:
I updated the sentence to include this. Let me know if there's anything else I could add there.